src/language/functions.md

```diff
@@ -500,6 +500,25 @@ Iterable<int> naturalsDownFrom(int n) sync* {
 }
 ```
 
+## External functions {#external}
+
+An external function is a function whose body is provided separately from its declaration.
+
+```dart
+external void someFunc(int i);
+```
+
+External functions can be top-level functions, [instance methods][],
+[getters or setters][], or [non-redirecting constructors][].
+[Instance variables][] can be `external` too,
+and essentially equate to a getter or setter on the external value. 
```

Comment:
We don't even have to say 'essentially' --- a final external variable is just a different syntax specifying an external getter, and a non-final external variable is just a different syntax specifying an external getter and an external setter. Also, we can't know whether there is anything out there that matches the description 'external value', it's perfectly possible that such getters/setters are bound to C functions, with no direct connection to anything that resembles a variable. Perhaps the text could be adjusted a bit like this:

```suggestion
An [instance variable][] can be `external` too,
which is equivalent to an external getter and (if the variable
is not `final`) an external setter.
```

It is also possible for a static variable to be external, and a library variable (aka a top-level variable), but not a local variable (and not a function parameter). I don't know if it's important to mention the static/library variants, that would depend on the level of detail/completeness which is customary and desired here.